lib/wallaroo/core/source/barrier_source/barrier_source.pony

```diff
@@ -339,3 +339,12 @@ actor BarrierSource is Source
 
   fun ref current_sequence_id(): SeqId =>
     _seq_id
+
+  ///////////////
+  // WATERMARKS
+  ///////////////
+  fun ref check_effective_input_watermark(current_ts: U64): U64 =>
```

Comment:
This is the identity function -- is it supposed to do any checking? Or is the name there in anticipation of its future use?

Reply:
This is an implementation of a `trait`.  On this source, the effective input watermark is always the current timestamp.